src/tiled/changepolygon.cpp

```diff
@@ -76,3 +76,50 @@ void TogglePolygonPolyline::toggle()
 {
     mMapObject->setShape((mMapObject->shape() == MapObject::Polygon) ? MapObject::Polyline : MapObject::Polygon);
 }
+
+SplitPolyline::SplitPolyline(MapDocument *mapDocument,
+                             MapObject *mapObject,
+                             int index)
+    : mMapDocument(mapDocument)
+    , mFirstPolyline(mapObject)
+    , mEdgeIndex(index)
+    , mObjectIndex(-1)
+{
+    mSecondPolyline = mFirstPolyline->clone();
+    mSecondPolyline->resetId();
+
+    setText(QCoreApplication::translate("Undo Commands", "Split Polyline"));
+}
+
+void SplitPolyline::undo()
+{
+    mObjectIndex = mMapDocument->mapObjectModel()->removeObject(mFirstPolyline->objectGroup(),
+                                                                mSecondPolyline);
+
+    QPolygonF firstPolygon = mFirstPolyline->polygon();
+    QPolygonF secondPolygon = mSecondPolyline->polygon();
+
+    firstPolygon.append(secondPolygon);
```

Comment:
Clang complains here:
~~~
changepolygon.cpp:102:25: error: no viable conversion from 'QPolygonF' to

      'const QPointF'

    firstPolygon.append(secondPolygon);

                        ^~~~~~~~~~~~~

/opt/qt54/include/QtCore/qpoint.h:205:21: note: candidate constructor (the

      implicit copy constructor) not viable: no known conversion from

      'QPolygonF' to 'const QPointF &' for 1st argument

class Q_CORE_EXPORT QPointF

                    ^

/opt/qt54/include/QtCore/qpoint.h:205:21: note: candidate constructor (the

      implicit move constructor) not viable: no known conversion from

      'QPolygonF' to 'QPointF &&' for 1st argument

class Q_CORE_EXPORT QPointF

                    ^

/opt/qt54/include/QtCore/qpoint.h:270:34: note: candidate constructor not

      viable: no known conversion from 'QPolygonF' to 'const QPoint &' for 1st

      argument

Q_DECL_CONSTEXPR inline QPointF::QPointF(const QPoint &p) : xp(p.x()), y...

                                 ^

/opt/qt54/include/QtGui/qpolygon.h:135:5: note: candidate function

    operator QVariant() const;

    ^

/opt/qt54/include/QtCore/qvector.h:130:26: note: passing argument to parameter

      't' here

    void append(const T &t);

                         ^

1 error generated.
~~~